RecoEgamma/EgammaElectronProducers/plugins/GsfElectronBaseProducer.cc

```diff
@@ -383,7 +383,7 @@ void GsfElectronBaseProducer::fillEvent( edm::Event & event )
    }
   // final filling
   auto finalCollection = std::make_unique<GsfElectronCollection>();
-  algo_->copyElectrons(*finalCollection) ;
+  algo_->moveElectrons(*finalCollection) ;
   orphanHandle_ = event.put(std::move(finalCollection));
```

Comment:
If `algo_` would return a moved-out `GsfElectronCollection` (I suppose the return type should then be `GsfElectronCollection&&`), and there would be a `edm::EDPutTokenT<GsfElectronCollection>`, these three lines could be collapsed to one (and avoid one heap allocation)
```cpp
orphanHandle_ = event.emplace(electronPutToken_, algo_->moveElectronsFrom());
```
(feel free to invent better method name)

(I'm not saying this needs to be done, but now that you're at it, the product could be put to the event also this way)